scripts/get-model-data.py

```diff
@@ -222,6 +236,105 @@
+        'model': 'briefs',
+        'joins': [
+            {
+                'model': 'awarded_brief_responses',
+                'left_on': 'id',
+                'right_on': 'briefId',
+                'data_duplicate_suffix': '_awarded_brief_responses'
+            }, {
+                'model': 'brief_responses',
+                'left_on': 'id',
+                'right_on': 'briefId',
+                'data_duplicate_suffix': '_data'
+            },
+
+        ],
+        'keys': (
+            'title',
+            'id_data',
+            'frameworkSlug',
+            'lotSlug',
+            'specialistRole',
+            'organisation',
+            'emailAddress',
+            'location',
+            'publishedAt',
+            'requirementsLength',
+            'totalApplications',
+            'applicationsFromSMEs',
+            'applicationsFromLargeOrganisations',
+            'status_briefs',
+            'awarded_supplierName',
+            'awarded_supplierOrganisationSize',
+            'awarded_awardedContractValue',
+            'awarded_awardedContractStartDate',
+        ),
+        'aggregation_counts': [
+            {
+                'group_by': 'id_data',
+                'join': ('id_data', 'id_data'),
+                'count_name': 'totalApplications',
+                'query': 'id_brief_responses != ""',
+            }, {
+                'group_by': 'id_data',
+                'join': ('id_data', 'id_data'),
+                'count_name': 'applicationsFromSMEs',
+                'query': 'supplierOrganisationSize in ["micro", "small", "medium"]',
+            }, {
+                'group_by': 'id_data',
+                'join': ('id_data', 'id_data'),
+                'count_name': 'applicationsFromLargeOrganisations',
+                'query': 'supplierOrganisationSize == "large"',
+            },
+        ],
+        'process_fields': {
+            'id_data': construct_brief_url,
+            'emailAddress': remove_username_from_email_address,
+        },
+        'rename_fields': {
+            'title': 'Opportunity',
+            'id_data': 'Link',
+            'frameworkSlug': 'Framework',
+            'lotSlug': 'Category',
+            'specialistRole': 'Specialist',
+            'organisation': 'Organisation Name',
+            'emailAddress': 'Buyer Domain',
+            'location': 'Location Of The Work',
+            'publishedAt': 'Published At',
+            'requirementsLength': 'Open For',
+            'applicationsFromSMEs': 'Applications from SMEs',
+            'applicationsFromLargeOrganisations': 'Applications from Large Organisations',
+            'totalApplications': 'Total Organisations',
+            'status': 'Status',
+            'awarded_supplierName': 'Winning supplier',
+            'awarded_supplierOrganisationSize': 'Size of supplier',
+            'awarded_awardedContractValue': 'Contract amount',
+            'awarded_awardedContractStartDate': 'Contract start date'
+        }
```

Comment:
One day (not today) I would love to see all this config held in a separate place to the code that churns out the CSVs themselves. We could have a lovely `ModelExportCSV` class that initialises with the config, queries the data, does some manipulation/renaming etc, has testable methods, avoids all that logic in the top level... _wanders off into dream land_

```
class ModelExportCSV(name, model, keys, joins=None, process_fields=None, ...):
    def get_query_data(self, ...):
          ...   
    def rename_fields(self, ...):
          ...  
    def output_to_dir(self, ...):
          ...  
    def run(self):
          # Do everything
          ...

opportunity_data_export = ModelExportCSV(**CONFIGS['opportunity_data'])
opportunity_data_export.run()
```

Something for the next 🔥 💔 maybe!

Reply:
Yeah, I'm gonna make a tech debt ticket for this